lib/elixir/lib/enum.ex

```diff
@@ -626,6 +626,74 @@ defmodule Enum do
     end)
   end
 
+  @doc """
+  Counts the enumerable stopping at `limit`.
+
+
+  If the enumerable implements `c:Enumerable.count/1`, this optimization is
+  not discarded. We simply call it and return the lower of the two numbers.
+  This means that `count_until/2` may not iterate over each item until count
+  is reached. To force enumeration, use `count_until/3`.
+
+  ## Examples
```

Comment:
Jose had some good examples in the thread for `is-at-least` and `is-exactly` scenarios that would be worth adding as examples. In particular, if you want to know that you have _exactly_ ten items, the fact that you need to `count_until(enum, 11)` is non-obvious.

Reply:
added